ml-agents/mlagents/trainers/ppo/trainer.py

```diff
@@ -332,34 +335,48 @@ def process_experiences(self, current_info: AllBrainInfo, new_info: AllBrainInfo
                 or len(agent_actions) > self.trainer_parameters["time_horizon"]
             ) and len(agent_actions) > 0:
                 agent_id = info.agents[l]
-                if info.local_done[l] and not info.max_reached[l]:
-                    value_next = 0.0
+                if info.max_reached[l]:
+                    bootstrapping_info = self.training_buffer[agent_id].last_brain_info
+                    idx = bootstrapping_info.agents.index(agent_id)
                 else:
-                    if info.max_reached[l]:
-                        bootstrapping_info = self.training_buffer[
-                            agent_id
-                        ].last_brain_info
-                        idx = bootstrapping_info.agents.index(agent_id)
-                    else:
-                        bootstrapping_info = info
-                        idx = l
-                    value_next = self.policy.get_value_estimate(bootstrapping_info, idx)
-
-                self.training_buffer[agent_id]["advantages"].set(
-                    get_gae(
-                        rewards=self.training_buffer[agent_id]["rewards"].get_batch(),
-                        value_estimates=self.training_buffer[agent_id][
-                            "value_estimates"
-                        ].get_batch(),
-                        value_next=value_next,
-                        gamma=self.trainer_parameters["gamma"],
+                    bootstrapping_info = info
+                    idx = l
+                value_next = self.policy.get_value_estimates(bootstrapping_info, idx)
+                if info.local_done[l] and not info.max_reached[l]:
+                    value_next["extrinsic"] = 0.0
+                tmp_advantages = []
+                tmp_returns = []
+                for idx, name in enumerate(self.policy.reward_signals.keys()):
```

Comment:
Don't need this `idx` or the `enumerate` (I think?)

Reply:
Yep, we're indexing only by name now - removed.